ui/src/main/java/org/pentaho/mantle/client/workspace/SchedulesPerspectivePanel.java

```diff
@@ -56,39 +58,65 @@ public SchedulesPerspectivePanel() {
     try {
       final String url = EnvironmentHelper.getFullyQualifiedURL() + "api/repo/files/canAdminister"; //$NON-NLS-1$
       RequestBuilder requestBuilder = new RequestBuilder( RequestBuilder.GET, url );
-      requestBuilder.setHeader( ACCEPT, ContentType.TEXT_PLAIN.toString() );
-      requestBuilder.setHeader( HttpHeaders.IF_MODIFIED_SINCE, IF_MODIFIED_SINCE );
+      requestBuilder.setHeader( ACCEPT, TEXT_PLAIN );
+      requestBuilder.setHeader( IF_MODIFIED_SINCE, IF_MODIFIED_SINCE_DATE );
       requestBuilder.sendRequest( null, new RequestCallback() {
 
         public void onError( Request request, Throwable caught ) {
           isAdmin = false;
           isScheduler = false;
+          canExecuteSchedules = false;
         }
 
         public void onResponseReceived( Request request, Response response ) {
           isAdmin = "true".equalsIgnoreCase( response.getText() ); //$NON-NLS-1$
 
-          try {
-            final String url2 = ScheduleHelper.getPluginContextURL() + "api/scheduler/canSchedule"; //$NON-NLS-1$
-            RequestBuilder requestBuilder2 = new RequestBuilder( RequestBuilder.GET, url2 );
-            requestBuilder2.setHeader( "accept", "application/json" ); //$NON-NLS-1$ //$NON-NLS-2$
-            requestBuilder2.setHeader( "If-Modified-Since", "01 Jan 1970 00:00:00 GMT" );
-            requestBuilder2.sendRequest( null, new RequestCallback() {
-
-              public void onError( Request request, Throwable caught ) {
-                isScheduler = false;
-                createUI();
-              }
-
-              public void onResponseReceived( Request request, Response response ) {
-                isScheduler = "true".equalsIgnoreCase( response.getText() ); //$NON-NLS-1$
-                createUI();
-              }
-
-            } );
-          } catch ( RequestException e ) {
-            Window.alert( e.getMessage() );
-          }
+          canSchedule();
+        }
+      } );
+    } catch ( RequestException e ) {
+      Window.alert( e.getMessage() );
+    }
+  }
+
+  private void canSchedule() {
+    try {
+      final String url = ScheduleHelper.getPluginContextURL() + "api/scheduler/canSchedule"; //$NON-NLS-1$
+      RequestBuilder requestBuilder = new RequestBuilder( RequestBuilder.GET, url );
+      requestBuilder.setHeader( ACCEPT, APPLICATION_JSON );
+      requestBuilder.setHeader( IF_MODIFIED_SINCE, IF_MODIFIED_SINCE_DATE );
+      requestBuilder.sendRequest( null, new RequestCallback() {
+        public void onError( Request request, Throwable caught ) {
+          isScheduler = false;
+          canExecuteSchedules();
+        }
+
+        public void onResponseReceived( Request request, Response response ) {
+          isScheduler = "true".equalsIgnoreCase( response.getText() ); //$NON-NLS-1$
+          canExecuteSchedules();
```

Comment:
it is redundant to check if the user can execute schedules when he already has the highest permission for managing schedules, which includes executing schedules